samples/azure-service-bus-netstandard/lock-renewal/sample.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Azure Service Bus Lock Renewal Sample
+summary: Long message processing with Azure Service Bus
+reviewed: 2021-02-22
+component: ASBS
+related:
+- transports/azure-service-bus
+- samples/azure/azure-service-bus-long-running
+---
+
+
+## Prerequisites
+
+include: asb-connectionstring-xplat
+
+<!-- include documentation from https://docs.particular.net/transports/azure-service-bus/legacy/message-lock-renewal -->
+
+## Important information about lock renewal
+
+1. Message lock renewal operation is initiated by the client code, not the broker. If the request to renew the lock fails after all the built into the SDK retries, the lock won't be re-acquired, and the message will become unlocked and available for re-processing by compteting consumers. Lock renewal should be treated as best-effort and not as a guaranteed operation.
+1. Message lock renewal applies to the currently processed message **only**. Prefetched messages that are not handled within the `LockDuration` time will lose their lock, indicated with a `LockLostException` in the log when they are attempted to be completed by the transport. To prevent the exception with lock renewal, prefetching should be adjusted to reduce number of prefetched files. Alternatively, endpoint's concurrency can be inclreased.
```

Comment:
```suggestion
1. Message lock renewal applies to the currently processed message **only**. Prefetched messages that are not handled within the `LockDuration` time will lose their lock, indicated with a `LockLostException` in the log when they are attempted to be completed by the transport. Prefetching can be adjusted to reduce the number of prefetched messages, which can help to prevent exceptions with lock renewal. Alternatively, the endpoint's concurrency can be increased to speed up the processing of messages due to the increased concurrency.
```